orchestrator/careplancontributor/service.go

```diff
@@ -104,24 +106,63 @@ func (s Service) confirm(localFHIR fhirclient.Client, serviceRequestRef string,
 	//if err != nil {
 	//	return nil, fmt.Errorf("failed to create CarePlan: %w", err)
 	//}
-	task, err := s.createTask(*serviceRequest, patient)
+	task, err := s.createTask(*serviceRequest)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create Task: %w", err)
 	}
-	// Add Task to CarePlan
-	// Note: since we created the Task with status=accepted, we can do this immediately.
-	// Otherwise, we have to poll/wait from the UI.
-	//carePlan.Activity = append(carePlan.Activity, fhir.CarePlanActivity{
-	//	Reference: &fhir.Reference{
-	//		Type:      to.Ptr("Task"),
-	//		Reference: to.Ptr("Task/" + *task.Id), // TODO: This seems a fiddly way to reference stuff
-	//	},
-	//})
-	//// TODO: Add "If" headers to make sure we're not overwriting someone else's changes
-	//if err := s.CarePlanService.Update("CarePlan/"+*carePlan.Id, carePlan, &carePlan); err != nil {
-	//	return nil, fmt.Errorf("failed to add Task to CarePlan: %w", err)
-	//}
-	return task, err
+
+	// Create channels for polling the task, 1 channel for the result and 1 for an error
+	resultChan := make(chan *fhir.Task)
+	errorChan := make(chan error)
+
+	// Start polling in a new goroutine so that the code continues to the select below
+	go s.pollTaskStatus(*task.Id, resultChan, errorChan)
+
+	// Block until either the task is accepted or an error occurs (error after 10 attempts)
+	select {
```

Comment:
this makes the call effectively blocking again, so no need to do `pollTaskStatus` in a goroutine